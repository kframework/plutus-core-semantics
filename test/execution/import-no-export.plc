(program

(module Bar (import) (export () ())
  (declare r1 (integer))
  (define  r1 (builtin multiplyInteger 3 7))
)

(module Baz (import Bar Foo) (export () ())
  (declare r2 (integer))
  (define  r2 (builtin multiplyInteger 2 Bar.r1))
)

)
