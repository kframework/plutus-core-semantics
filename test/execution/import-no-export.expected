<T>
  <k> ( Bar . r1 ) ~> #freezer(builtin___)0 ( multiplyInteger , 2 ) ~> #freezer(define__)0 ( r2 ) ~> .DecList ~> ( program .ModList ) </k>
  <lenv> .Map </lenv>
  <genv> ( Bar . r1 ) |-> 21 </genv>
  <modules>
    <module>
      <curr> 1 </curr>
      <name> Baz </name>
      <data> .TyExpList </data>
      <decl> r2  .NameList </decl>
      <import> Bar  .ModNmList </import>
      <exportTy> .TyExpList </exportTy>
      <exportTm> .NameList </exportTm>
    </module>
    <module>
      <curr> 0 </curr>
      <name> Bar </name>
      <data> .TyExpList </data>
      <decl> r1  .NameList </decl>
      <import> .ModNmList </import>
      <exportTy> .TyExpList </exportTy>
      <exportTm> .NameList </exportTm>
    </module>
  </modules>
</T>
